validator/rpc/wallet.go

```diff
@@ -165,6 +133,78 @@ func (s *Server) WalletConfig(ctx context.Context, _ *empty.Empty) (*pb.WalletRe
 	}, nil
 }
 
+// RecoverWallet via an API request, allowing a user to recover a derived.
+// Generate the seed from the mnemonic + language + 25th passphrase(optional).
+// Create N validator keystores from the seed specified by req.NumAccounts.
+// Set the wallet password to req.WalletPassword, then create the wallet from
+// the provided Mnemonic and return CreateWalletResponse,nil if nothing went wrong
+func (s *Server) RecoverWallet(ctx context.Context, req *pb.RecoverWalletRequest) (*pb.CreateWalletResponse, error) {
+	numAccounts := int(req.NumAccounts)
+	if numAccounts < 1 {
+		return nil, status.Error(codes.InvalidArgument, "Must create at least 1 validator account")
+	}
+
+	// Check validate mnemonic with chosen language
+	language := req.Language
+	allowedLanguages := map[string][]string{
+		"english":             wordlists.English,
+		"chinese_simplified":  wordlists.ChineseSimplified,
+		"chinese_traditional": wordlists.ChineseTraditional,
+		"french":              wordlists.French,
+		"italian":             wordlists.Italian,
+		"japanese":            wordlists.Japanese,
+		"korean":              wordlists.Korean,
+		"spanish":             wordlists.Spanish,
+	}
+	if _, ok := allowedLanguages[language]; !ok {
+		return nil, status.Error(codes.InvalidArgument, "input not in the list of supported languages")
+	}
+	bip39.SetWordList(allowedLanguages[language])
+	mnemonic := req.Mnemonic
+	if err := accounts.ValidateMnemonic(mnemonic); err != nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid mnemonic in request")
+	}
+	if !req.SkipMnemonic25ThWord && strings.TrimSpace(req.Mnemonic25ThWord) == "" {
+		return nil, status.Error(codes.InvalidArgument, "mnemonic 25th word cannot be empty")
+	}
+
+	// Web UI is structured to only write to the default wallet directory
+	// accounts.Recoverwallet checks if wallet already exists
+	walletDir := s.walletDir
+
+	// Web-ui should check the new and confirmed password are equal
+	walletPassword := req.WalletPassword
+	if err := promptutil.ValidatePasswordInput(walletPassword); err != nil {
+		return nil, status.Error(codes.InvalidArgument, "password did not pass validation")
+	}
+
+	if _, err := accounts.RecoverWallet(ctx, &accounts.RecoverWalletConfig{
+		WalletDir:        walletDir,
+		WalletPassword:   walletPassword,
+		Mnemonic:         mnemonic,
+		NumAccounts:      numAccounts,
+		Mnemonic25thWord: req.Mnemonic25ThWord,
+	}); err != nil {
+		return nil, err
+	}
+	if err := s.initializeWallet(ctx, &wallet.Config{
+		WalletDir:      walletDir,
+		KeymanagerKind: keymanager.Derived,
+		WalletPassword: walletPassword,
+	}); err != nil {
+		return nil, err
+	}
+	if err := writeWalletPasswordToDisk(walletDir, walletPassword); err != nil {
```

Comment:
why are writing the password to disk ? Isn't this something we are moving away from 
cc @rauljordan

Reply:
I believe we write the password to disk if a feature flag is enabled to help companies such as dappnode start their validator from the password file if needed. It does not get saved for regular users unless they add this flag. Since this conditional is hidden within the helper function, it looks opaque from the perspective of the reader, so perhaps we should do something like:

```go
if featureconfig.Get().WriteWalletPasswordToDisk {
  writeWalletPasswordToDisk
}
```
Up to you if you want to do that change in this PR @ahadda5

Reply:
But that is already part of the function itself. 
if  !writeWalletPassowrdonWebOnBoarding  don't write.   Still need it?

Reply:
@rauljordan  check above [Discussion](https://github.com/prysmaticlabs/prysm/pull/8679#discussion_r603139561)